### PR TITLE
add python2.6 compatibility

### PR DESCRIPTION
**Pull request by fstavela**

### CODE REVIEW

**Comment by fstavela** [comment]
Would it be acceptable to add Python2.6 compatibility? We need it for RHEL6 testing. If not, what was the last version that was compatible with Python2.6?

File: betelgeuse/__init__.py [hunk 1]
```diff
@@ -352,7 +362,9 @@ def create_xml_testcase(config, testcase, automation_script_format):
             testcase.docstring = testcase.docstring.decode('utf8')
 
     # Check if any field needs a default value
-    testcase.fields = {k.lower(): v for k, v in testcase.fields.items()}
+    for k, v in testcase.fields.items():
+        testcase.fields.pop(k)
```
Comment (mirekdlugosz) [inline betelgeuse/__init__.py]: In Python 2, `items()` returns copy of dict `key, value` pairs, but in Python 3 it returns special object that reflects changes in dict dynamically. Old code created new dict that we assigned to the same variable name as old one, now we modify dict in place. Iterating and modifying at the same time doesn't sound like a good idea to me, and Python 3 docs explicitly warns against it:

> Iterating views while adding or deleting entries in the dictionary may raise a RuntimeError or fail to iterate over all entries.

**Review by mirekdlugosz — COMMENTED** [review]
From code perspective, this is rather small and self-contained. I put comment in one place where I believe code has unintended consequence.

On the other hand, we can't support Python 2 indefinitely and we are probably past the cutting point already. I'm not sure if re-introducing Python 2 support now is a thing we should do. Luckily I am not maintainer here and it's up to @elyezer to decide.

Betelgeuse 1.0.0 is first version that is Python 3 exclusive, meaning 0.15.0 is the last that supports Python 2 (but I'm not sure if it supports Python 2.6).

File: betelgeuse/__init__.py (hunk 1 above)
Comment (fstavela) [reply betelgeuse/__init__.py]: I changed this. Now it creates new dictionary and then assigns it to the testcase.fields.

**Comment by elyezer** [comment]
Betelgeuse dropped support to Python 2 when the version 1 was released. The expected workflow is to not run it on RHEL 6 but another machine with Python 3.6+. Betelgeuse uses the jUnit XML file as the input so the testing can still happen on RHEL6, then you collect the jUnit XML and run Betelgeuse on a supported Python version.

Also we use pytest to run tests and flake8 for linting the source code and ensuring its quality. That said we would need to also update the tooling to support it. Furthermore we would not have the ability to run the tests on travis (see https://docs.travis-ci.com/user/languages/python/#specifying-python-versions) since it does not have Python 2.6 available.

With all that said, if you really want to have that support and can't change the workflow then I would suggest adding it on the Betelgeuse 0.x series rather than on 1.x which has dropped Python 2 support.

**Comment by elyezer** [comment]
Closing this based on the above conversation

**Pull request closed without merging**